server/src/main/java/org/opensearch/common/io/stream/NamedWriteableRegistryParseRequest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.io.stream;
+
+import org.opensearch.transport.TransportRequest;
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Extensibility support for Named Writeable Registry: request to extensions to parse context
+ *
+ * @opensearch.internal
+ */
+public class NamedWriteableRegistryParseRequest extends TransportRequest {
+
+    private String categoryClassName;
+    private byte[] context;
+
+    public NamedWriteableRegistryParseRequest(String categoryClassName, byte[] context) {
+        this.categoryClassName = categoryClassName;
+        this.context = context;
+    }
+
+    public NamedWriteableRegistryParseRequest(StreamInput in) throws IOException {
+        super(in);
+        this.categoryClassName = in.readString();
+        this.context = in.readByteArray();
+    }
+
+    public Class<?> getCategoryClass() throws ClassNotFoundException {
```

Comment:
Wildcard type? Can we have a specific type here for the Class?

Reply:
Looking at https://github.com/opensearch-project/opensearch-sdk/pull/54, it seems the type should be `NamedWriteable`?

Reply:
Category Classes are expected to extend the NamedWriteable class, I'll change all wildcard type references to reflect this.